content/post/new-python-api.md

```diff
@@ -0,0 +1,263 @@
++++
+title= "Idiomatic Python Stream Processing in Wallaroo"
+date = 2018-02-01T12:00:00-04:00
+draft = false
+author = "aturley"
+description = "Wallaroo’s New Python API makes writing distributed stream processing applications more Pythonic."
+tags = [
+    "wallaroo",
+    "python",
+    "api"
+]
+categories = [
+    "Python API"
+]
++++
+
+We have been working on Wallaroo, our scale-independent event processing system, for a little over two years. When we open sourced it in September of 2017 we included an API for writing applications using a Python API. This blog post tells the story of what we learned from the feedback we received about the original API and how we applied that feedback to make improvements that have led to our new API.
+
+Three months ago I wrote [a blog post](https://blog.wallaroolabs.com/2017/10/go-python-go-stream-processing-for-python/) about the first iteration of our Python API. At the end of that post I wrote, "Our API is new, and we are looking at ways to improve it. We have a lot of ideas of our own, but if you have any ideas, we would love to hear from you." We got some feedback on various channels, including [Hacker News](https://news.ycombinator.com/item?id=15457343) and direct communications with folks who were interested in Wallaroo. One theme that emerged from these conversations was that folks felt that the API didn’t use Python in an idiomatic way (a quality that Python users often refer to as “Pythonic”). And internally we had felt that there were things that we could improve in the API.
+
+We took the feedback and began to think about what kinds of things made the existing API unpythonic. Then we mocked up some of our ideas and tried them out by reimplementing some of our example programs to see how they felt. Finally, we worked on solving some unexpected problems that arose from the changes. The end result is a new Wallaroo Python API that is much more concise and Pythonic than the original. We’ve gotten some great early feedback, and we think it represents a notable improvement over the original API, and we think that Python programmers will agree.
```

Comment:
The two uses of `and we think` sounds a bit awkward. Maybe rephrase to:

> We’ve gotten some great early feedback indicating a notable improvement over the original API, and we think that Python programmers will agree.

Reply:
Removing second one.